BootStrapTests#testTriggeredWatchLoading can fail assertion

**Build scan**: https://gradle-enterprise.elastic.co/s/nnardp2pu6tau

**Repro line**: 
```
./gradlew ':x-pack:plugin:watcher:internalClusterTest' \
  --tests "org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.testTriggeredWatchLoading" \
  -Dtests.seed=7DB0FF08E8B2DB2C -Dtests.security.manager=true -Dtests.locale=it-CH \
  -Dtests.timezone=Europe/Samara -Druntime.java=11
```

**Reproduces locally?**: no

**Applicable branches**: master (at least)

**Failure history**: fails sporadically but regularly, around twice a month

**Failure excerpt**:
```
 2> java.lang.AssertionError: Count is 8 hits but 4 was expected.  Total shards: 1 Successful shards: 1 & 0 shard failures:
        at __randomizedtesting.SeedInfo.seed([7DB0FF08E8B2DB2C:445E9FB2D758E3D3]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount(ElasticsearchAssertions.java:270)
        at org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.lambda$assertSingleExecutionAndCompleteWatchHistory$3(BootStrapTests.java:288)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:955)
        at org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.assertSingleExecutionAndCompleteWatchHistory(BootStrapTests.java:274)
        at org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.testTriggeredWatchLoading(BootStrapTests.java:269)
```

A more recent failure: https://gradle-enterprise.elastic.co/s/hspurs4o6yl3g
Failed twice in the last two weeks.

```
org.elasticsearch.xpack.watcher.test.integration.BootStrapTests > testTriggeredWatchLoading FAILED
    java.lang.AssertionError: Count is 10 hits but 5 was expected.  Total shards: 1 Successful shards: 1 & 0 shard failures:
        at __randomizedtesting.SeedInfo.seed([F4277DDFA8895231:CDC91D6597636ACE]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount(ElasticsearchAssertions.java:258)
        at org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.lambda$assertSingleExecutionAndCompleteWatchHistory$3(BootStrapTests.java:289)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:977)
        at org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.assertSingleExecutionAndCompleteWatchHistory(BootStrapTests.java:275)
        at org.elasticsearch.xpack.watcher.test.integration.BootStrapTests.testTriggeredWatchLoading(BootStrapTests.java:270)
```

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/BootStrapTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/BootStrapTests.java
@@ -259,6 +259,7 @@ public class BootStrapTests extends AbstractWatcherIntegrationTestCase {
     }
 
     public void testTriggeredWatchLoading() throws Exception {
+        cluster().wipeIndices(TriggeredWatchStoreField.INDEX_NAME);
         createIndex("output");
         client().prepareIndex()
             .setIndex("my-index")

